src/storage.c

```diff
@@ -1,15 +1,22 @@
 #include "storage.h"
+#include <string.h>
 
-uint32_t get_secret() {
-    if (N_storage.secret == 0) {
-        // first access, generate and write to storage
-        return generate_secret();
+#include "os.h"
+
+void get_secret(uint8_t* secret) {
+    for (uint8_t i = 0; i < SECRET_LEN; i++) {
+        if (*((volatile uint8_t*) N_storage.secret + i) != 0) {
```

Comment:
What is volatile used for here?

Reply:
You're right, there is no need to use volatile here